src/app/components/import-annotations/details/details.component.ts

```diff
@@ -0,0 +1,333 @@
+    this.performDryRun(model);
+  }
+
+  // a predicate to check if every group in the buffer is valid
+  // this is used for form validation
+  protected isBufferValid(): boolean {
+    const importErrors = this.importBuffer.flatMap((model) => model?.errors);
+    return this.importBuffer.length > 1 && importErrors.length === 0;
+  }
+
+  protected removeFromBuffer(index: number): void {
+    this.importBuffer.splice(index, 1);
+  }
+
+  // sends all valid imports in the buffer to the api
+  protected submitImportBuffer(): void {
+    // importing invalid annotation imports results in an internal server error
+    // we should therefore filter out any invalid models before importing
+    // however, this should be caught by form validation, preventing a user to import if there are errors
+    const validImportBuffers = this.importBuffer.filter(
+      (model) => model.errors.length === 0 && model.files?.length > 0
+    );
+
+    validImportBuffers.forEach((model: ImportGroup, i: number) =>
+      this.importEventGroup(model, i)
+    );
+  }
+
+  protected importEventGroup(model: ImportGroup, i: number): void {
+    // we have to do this because by default, a FileList is not an array, but uses an iterator
+    // therefore, if we convert it to an array, we can use a for-loop to import each file individually
+    const modelFiles: File[] = Array.from(model.files);
+
+    modelFiles.forEach((file: File) => {
+      const audioEventImportModel: AudioEventImportFileWrite =
+        new AudioEventImportFileWrite({
+          id: this.audioEventImport.id,
+          file,
+          additionalTagIds: model.additionalTagIds,
+          commit: true,
+        });
+
+      // because the event imports api implement the baw-api service, we don't need to catch errors here
+      this.eventImportsApi
+        .importFile(audioEventImportModel)
+        .pipe(takeUntil(this.unsubscribe))
+        .subscribe({
+          next: () => {
+            this.removeFromBuffer(i);
+            this.refreshTables();
+          },
+          error: (error: BawApiError) => {
+            // some of the default error messages are ambiguous on this page
+            // e.g. "you do not have access to this page" means that the user doesn't have access to the audio recording
+            if (error.status === FORBIDDEN) {
+              this.notifications.error(
+                "You do not have access to all the audio recordings or tags in your files."
+              );
+            } else {
+              return throwError(() => error);
+            }
+          },
+        });
+    });
+  }
+
+  // since the typeahead input returns an array of models, but the api wants the associated tags as an array of id's
+  // we use this helper function to convert the array of models to an array of id's that can be sent in the api request
+  protected getIdsFromAbstractModelArray(items: object[]): Id[] {
+    return items.map((item: AbstractModel): Id => item.id);
+  }
```

Comment:
This function is needed because the typeahead-input returns a list of tag models, while the event import requires an array of tag id's.

I don't know how I'd use a model association for this??

Reply:
oh right. Our associations code translates an array of ids to an array of resolved objects.

When we update an array of resolved objects, we should update the original array of ids automatically. 
Thus, when the serialization happens the ids are already there.

~Might be out of scope of this PR~ is out of scope. Our associations are one-way only. nvm